### PR TITLE
 ✨ clusterctl: print stack trace if verbosity is >= 5

### PR DESCRIPTION
**Pull request by cpanato**
**What this PR does / why we need it**:
remove `// TODO: print error stack if log v>0`

print that stack trace when have any error and `-v`or `CLUSTERCTL_LOG_LEVEL` is set

example:

```
➜ ./bin/clusterctl config cluster my -v5
Error: failed to identify the default infrastructure provider. Please specify an infrastructure provider: unsupported management cluster server version: v1.14.6 - minimum required version is v1.16.0
sigs.k8s.io/cluster-api/cmd/clusterctl/client.(*clusterctlClient).getTemplateFromRepository
        /Users/cpanato/go/src/github.com/kubernetes-sigs/cluster-api/cmd/clusterctl/client/config.go:209
sigs.k8s.io/cluster-api/cmd/clusterctl/client.(*clusterctlClient).GetClusterTemplate
        /Users/cpanato/go/src/github.com/kubernetes-sigs/cluster-api/cmd/clusterctl/client/config.go:189
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.runGetClusterTemplate
        /Users/cpanato/go/src/github.com/kubernetes-sigs/cluster-api/cmd/clusterctl/cmd/config_cluster.go:187
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.glob..func1
        /Users/cpanato/go/src/github.com/kubernetes-sigs/cluster-api/cmd/clusterctl/cmd/config_cluster.go:92
github.com/spf13/cobra.(*Command).execute
        /Users/cpanato/go/pkg/mod/github.com/spf13/cobra@v0.0.6/command.go:840
github.com/spf13/cobra.(*Command).ExecuteC
        /Users/cpanato/go/pkg/mod/github.com/spf13/cobra@v0.0.6/command.go:945
github.com/spf13/cobra.(*Command).Execute
        /Users/cpanato/go/pkg/mod/github.com/spf13/cobra@v0.0.6/command.go:885
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.Execute
        /Users/cpanato/go/src/github.com/kubernetes-sigs/cluster-api/cmd/clusterctl/cmd/root.go:52
main.main
        /Users/cpanato/go/src/github.com/kubernetes-sigs/cluster-api/cmd/clusterctl/main.go:25
runtime.main
        /usr/local/Cellar/go/1.14.2_1/libexec/src/runtime/proc.go:203
runtime.goexit
        /usr/local/Cellar/go/1.14.2_1/libexec/src/runtime/asm_amd64.s:1373


➜ ./bin/clusterctl config cluster my
Error: failed to identify the default infrastructure provider. Please specify an infrastructure provider: unsupported management cluster server version: v1.14.6 - minimum required version is v1.16.0
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

/cc @wfernandes @fabriziopandini 

/kind cleanup

### CODE REVIEW

File: cmd/clusterctl/cmd/root.go [hunk 1]
```diff
@@ -45,7 +50,13 @@ var RootCmd = &cobra.Command{
 
 func Execute() {
 	if err := RootCmd.Execute(); err != nil {
-		// TODO: print error stack if log v>0
+		if *verbosity >= 5 {
```
Comment (vincepri) [inline cmd/clusterctl/cmd/root.go:53]: Should we also check that verbosity is not nil?

**Comment by wfernandes** [comment]
Reviewing.

File: cmd/clusterctl/cmd/root.go (hunk 1 above)
Comment (wfernandes) [reply cmd/clusterctl/cmd/root.go:53]: +1 Guess it doesn't hurt to do a nil check even though the default is set to 0.

Comment (cpanato) [reply cmd/clusterctl/cmd/root.go:53]: done

Comment (cpanato) [reply cmd/clusterctl/cmd/root.go]: PTAL when having some time :)

**Pull request merged**